tools/wptrunner/wptrunner/executors/executorchrome.py

```diff
@@ -94,6 +96,31 @@ def get_counters(self) -> Mapping[str, int]:
         return counters
 
 
+class ChromeDriverTestDriverProtocolPart(WebDriverTestDriverProtocolPart):
+    """An interface to the browser-side testdriver infrastructure that lazily settles calls."""
+
+    def setup(self):
+        super().setup()
+        self._pending_message = None
+
+    def send_message(self, cmd_id, message_type, status, message=None):
+        message_script = self._format_send_message_script(cmd_id, message_type, status, message)
+        if message_type == "complete":
+            message_script = self._pop_message(message_script)
+        if message_script:
+            self.webdriver.execute_script(message_script)
+
```

Comment:
Can we use the below?
```
if message_type == "complete":
    self._pending_message = message_script
else:
    self.webdriver.execute_script(message_script)
```

Then we do not need `_pop_message`. That is to say all the 'complete' message will be pigpacked with resume script, and no change to other message.

Reply:
Done; I think that's fine to achieve some code simplification. Consecutive `send_message("complete")` without `get_next_message()` between will drop all but the last message, but that doesn't really make sense for a caller to do. Added an `assert` to hard-fail just in case that's violated.